code/game/objects/items/RPD.dm

```diff
@@ -561,20 +556,26 @@ GLOBAL_LIST_INIT(RPD_recipes, list(
 	. = FALSE
 	switch(p_class) //if we've gotten this var, the target is valid
 		if(PAINT_MODE) //Paint pipes
+			if(!istype(A, /obj/machinery/atmospherics/pipe))
+				return ..()
 			var/obj/machinery/atmospherics/pipe/P = A
 			playsound(get_turf(src), 'sound/machines/click.ogg', 50, 1)
 			P.paint(paint_colors[paint_color])
 			user.visible_message("<span class='notice'>[user] paints \the [P] [paint_color].</span>","<span class='notice'>You paint \the [P] [paint_color].</span>")
 			return
 
 		if(EATING_MODE) //Eating pipes
+			if(!((istype(A, /obj/item/pipe) || istype(A, /obj/item/pipe_meter) || istype(A, /obj/structure/disposalconstruct))))
```

Comment:
`if(!(istype(A, /obj/item/pipe) || istype(A, /obj/item/pipe_meter) || istype(A, /obj/structure/disposalconstruct)))`
you left an extra pair of parentheses; my line removes them